### PR TITLE
WIP: python-CodeIntel-2.0.0

### PR DESCRIPTION
**Pull request by shizonic**
I need some help finishing the template. Currently when I install the `python2` version I get 3 binaries (codeintel, codeintel2, and codenintel3) and when I install the `python3` version I only get one binary (codeintel3).

Further I thnink there are some more dependencies. But I am not that familiar with the setuptools build tool and therefore I can't find out which dependencies are necessary.

Does anybody can help me?

### CODE REVIEW

File: srcpkgs/python-CodeIntel/template [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+pycompile_module="codeintel"
+short_desc="Code intelligence engine that was ported from Open Komodo
+ Editor to Python2"
+maintainer="shizonic <realtiaz@gmail.com>"
+homepage="https://github.com/SublimeCodeIntel/CodeIntel"
+license="MPL"
+distfiles="${PYPI_SITE}/C/CodeIntel/CodeIntel-${version}.tar.gz"
+checksum=0c3d0fcd356082b0f786e52151e36e2ca8622ea399896ac5478d39e7f78815e8
+
+pkg_install() {
```
Comment (Johnnynator) [inline srcpkgs/python-CodeIntel/template]: pkg_install is only fur subpackages

File: srcpkgs/python-CodeIntel/template [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	vmove usr/lib/python2*
+	vbin usr/bin/codeintel2
+	vlicense LICENSE.txt
+}
+
+python3-CodeIntel_package() {
+	noarch=yes
+	makedepends="python3-devel"
```
Comment (Johnnynator) [inline srcpkgs/python-CodeIntel/template]: subpackages don't have they own makedepends.

**Comment by shizonic** [comment]
@Johnnynator Another question: The `python2` version builds successfully but depends on `clang 5.x`. So does it make sense to build it anyway or should I just provide a `python3` package?

**Comment by Johnnynator** [comment]
> @Johnnynator Another question: The `python2` version builds successfully but depends on `clang 5.x`. So does it make sense to build it anyway or should I just provide a `python3` package?

If it depends on clang being version 5 then it is useless. Yes if you want you can go python3 only, there is no need for new python2 packages, since the current state is to try to migrate to python3 only.

Also python3 seems to be somewhat broken when looking at travis:

```
reading manifest template 'MANIFEST.in'
  File "/usr/lib/python3.6/site-packages/codeintel/ciperf.py", line 148
    print "load directory '%s'" % perllib
                              ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("load directory '%s'" % perllib)?
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/broken_on_curr_line.py", line 12
    trentm.        # current line is here, cursor is meant to be at the '.'
                                                                          ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/with_stmt.py", line 8
    print line
             ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(line)?
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/invalid_arg_list.py", line 1
SyntaxError: positional argument follows keyword argument
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/eol_LF.py", line 9
    print "%s i=%d" % (name, i)
                  ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/eol_CR.py", line 9
    print "%s i=%d" % (name, i)
                  ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/eol_CRLF.py", line 9
    print "%s i=%d" % (name, i)
                  ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/lower_ord_chars.py", line 74
    print __doc__
                ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(__doc__)?
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/try_stmt.py", line 68
    print line
             ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(line)?
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/policy.py", line 38
    raise serverExceptions.PolicyViolation, \
                                          ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/python/xmlreader.py", line 298
    raise KeyError, name
                  ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/scan_inputs/py3/exceptions.py", line 6
    except Foo, Bar as msg:
              ^
SyntaxError: invalid syntax
  File "/usr/lib/python3.6/site-packages/codeintel/test2/bits/foo.py", line 20
    print "this is ", os.path.abspath(__file__)
                   ^
SyntaxError: invalid syntax
warning: no files found matching '*.rst' anywhere in distribution
```

File: srcpkgs/python-CodeIntel/template [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'python-CodeIntel'
+pkgname=python-CodeIntel
+version=2.0.0
+revision=1
+noarch=yes
```
Comment (Johnnynator) [inline srcpkgs/python-CodeIntel/template]: It is shipping with `*.so` so files so it is architecture dependent. remove noarch.

Comment (shizonic) [reply srcpkgs/python-CodeIntel/template]: Okay did remove it. Thanks

**Comment by shizonic** [comment]
@Johnnynator I will create a new PR for the python3-only package and close this one. I recognized the python3 error too but can't figure out how to fix it... Any ideas?

**Pull request closed without merging**